### PR TITLE
Feat/cluster

### PR DESCRIPTION
**Pull request by Tanguylo**
* **What kind of changes does this PR introduce?** 
- [x] new features

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
- [x] Yes: 
       - New subclasses of DessiaObject : HeterogeneousList(DessiaObject) and CategorizedList(HeterogeneousList)
       - No changes required in users' applications
       - New practices for handling lists of data and creating subsets of datasets

* **Other information**:
In next releases, HeterogeneousLists will be equiped with all identified datas mining useful technologies

### CODE REVIEW

**Review by masfaraud — CHANGES_REQUESTED** [review]

File: dessia_common/templates.py [hunk 1]
```diff
@@ -320,3 +320,16 @@
 </body>
 </html>
 ''')
+
+
+heterogeneouslist_markdown_template = Template('''
```
[the remark this answers is not in the document]
Comment (Tanguylo) [reply dessia_common/templates.py:325]: Removed for the moment.
I made this to prepare the markdown implementation.

File: dessia_common/tests.py [hunk 2]
```diff
@@ -177,25 +178,218 @@ def __init__(self, name: str, mpg: float, cylinders: float,
...
+        Car.__init__(self, name, mpg, cylinders, displacement, horsepower,
+                     weight, acceleration, model, origin)
+
+    @classmethod
+    def vector_features(cls):
+        return cls._vector_features
 
-        return cars, variables
+
+class ClusTesterD1(DessiaObject):
```
[the remark this answers is not in the document]
Comment (Tanguylo) [reply dessia_common/tests.py:225]: Changed to RandDataDn (n in range(10))

File: dessia_common/workflow/blocks.py [hunk 3]
```diff
@@ -354,6 +354,37 @@ def evaluate(self, values):
         return [[values[var] for var in self.inputs]]
 
 
+class Concatenate(Block):
```
[the remark this answers is not in the document]
Comment (Tanguylo) [reply dessia_common/workflow/blocks.py:357]: Done for dict (uncertain behavior) and HLists.
Any idea for other type ?

File: tests/_simulation_workflow.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import dessia_common.workflow as dcw
+import dessia_common.workflow.blocks as dcw_blocks
+import dessia_common.typings as dct
```
[the remark this answers is not in the document]
Comment (Tanguylo) [reply tests/_simulation_workflow.py:3]: file removed because I don't know what it is

**Review by masfaraud — APPROVED** [review]

**Review by masfaraud — APPROVED** [review]

**Review by masfaraud — APPROVED** [review]

**Pull request merged**